### PR TITLE
soc: arm: st_stm32: add STM32H725XX family support

### PR DESCRIPTION
**Pull request by aunsbjerg**
Add support for the stm32h725 family.

The only difference between the stm32h723 and stm32h725 families are that the h725 family has an built-in SMPS module which is available for use in code only when the `STM32H725XX` define is set, which this PR provides.

Changes have been tested on a custom board using an STM32H725VG mcu.

### CODE REVIEW

**Comment by manoj153** [comment]
Quickly ruining through the changes, I don't see anywhere you're enabling SMPS.

**Comment by erwango** [comment]
> Quickly ruining through the changes, I don't see anywhere you're enabling SMPS.

@manoj153 SMPS support is declared in CMSIS file: `stm32cube/stm32h7xx/soc/stm32h725xx.h`.

File: dts/arm/st/h7/stm32h725.dtsi [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/*
+ * Copyright (c) 2021 Esco Medical ApS
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/h7/stm32h723.dtsi>
```
Comment (erwango) [inline dts/arm/st/h7/stm32h725.dtsi:6]: It would be worth adding a comment stating: STM32H725 is STM32H723 + SMPS

**Review by erwango — CHANGES_REQUESTED** [review]
LGTM aside a comment that would be worth adding in device tree.

**Comment by manoj153** [comment]
> > Quickly ruining through the changes, I don't see anywhere you're enabling SMPS.
> 
> @manoj153 SMPS support is declared in CMSIS file: `stm32cube/stm32h7xx/soc/stm32h725xx.h`.

If I understood right, @aunsbjerg want to use SMPS, so I suppose `CONFIG_POWER_SUPPLY_DIRECT_SMPS` should be set, which seems to be missing now.

**Comment by aunsbjerg** [comment]
> > > Quickly ruining through the changes, I don't see anywhere you're enabling SMPS.
> > 
> > 
> > @manoj153 SMPS support is declared in CMSIS file: `stm32cube/stm32h7xx/soc/stm32h725xx.h`.
> 
> If I understood right, @aunsbjerg want to use SMPS, so I suppose `CONFIG_POWER_SUPPLY_DIRECT_SMPS` should be set, which seems to be missing now.

You are right, I do want to use SMPS and I do use `CONFIG_POWER_SUPPLY_DIRECT_SMPS=y` in my board definition. I do not know enough about the SoC to judge if SMPS should be enabled by default, but if that is the case, then I will of course add it to the appropriate kconfig file.

**Comment by manoj153** [comment]
> > > > Quickly ruining through the changes, I don't see anywhere you're enabling SMPS.
> > > 
> > > 
> > > @manoj153 SMPS support is declared in CMSIS file: `stm32cube/stm32h7xx/soc/stm32h725xx.h`.
> > 
> > 
> > If I understood right, @aunsbjerg want to use SMPS, so I suppose `CONFIG_POWER_SUPPLY_DIRECT_SMPS` should be set, which seems to be missing now.
> 
> You are right, I do want to use SMPS, and I do use `CONFIG_POWER_SUPPLY_DIRECT_SMPS=y` in my board definition. I do not know enough about the SoC to judge if SMPS should be enabled by default, but if that is the case, then I will of course, add it to the appropriate kconfig file.

Sorry, you're doing it correctly. I just realised this PR is related to SoC.

**Comment by aunsbjerg** [comment]
> Sorry, you're doing it correctly. I just realised this PR is related to SoC.

No worries :)

@erwango I pushed an updated dts commit with a comment on the reason for inheriting the h723 devicetree, as you suggested.

**Review by erwango — DISMISSED** [review]

File: dts/arm/st/h7/stm32h725Xg.dtsi [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#include <mem.h>
+#include <st/h7/stm32h725.dtsi>
+
+/ {
+	soc {
+		flash-controller@52002000 {
+			flash0: flash@8000000 {
+				compatible = "soc-nv-flash";
+				label = "FLASH_STM32";
+				reg = <0x08000000 DT_SIZE_K(1024)>;
```
Comment (MaureenHelm) [inline dts/arm/st/h7/stm32h725Xg.dtsi:15]: You can use `DT_SIZE_M(1)` instead

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
small nitpick

**Comment by erwango** [comment]
@MaureenHelm Can you have a new look ?

**Review by FRASTM — APPROVED** [review]

**Comment by aunsbjerg** [comment]
@erwango @MaureenHelm anything I can do to get this PR merged?

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**